### PR TITLE
fix: expand and collapse whole folder groups

### PR DESCRIPTION
**Pull request by Danie-1**
Currently when opening folder groups, the folder icon and arrow indicators don't change correctly.
Current behaviour:
image
With this fix:
image

### CODE REVIEW

**Comment by alex-courtis** [comment]
Regression since https://github.com/nvim-tree/nvim-tree.lua/pull/2100

File: lua/nvim-tree/lib.lua [hunk 1]
```diff
@@ -81,6 +81,16 @@ function M.get_last_group_node(node)
   return next
 end
 
+function M.get_all_nodes_in_group(node)
```
Comment (alex-courtis) [inline lua/nvim-tree/lib.lua:84]: We could simplify a bit `open_all_nodes_in_group` howeve this function will likely see use in the future.

File: lua/nvim-tree/lib.lua [hunk 2]
```diff
@@ -81,6 +81,16 @@ function M.get_last_group_node(node)
...
+  local next = utils.get_parent_of_group(node)
```
Comment (alex-courtis) [inline lua/nvim-tree/lib.lua:85]: Please don't overload the builtin function `next`.

I know it's been used elsewhene...

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Many thanks for addressing this one!

- [x] replace `next`
- [x] bonus points: replace other usages of `next`

**Comment by alex-courtis** [comment]
When removing the last file in a grouped or ungrouped open folder it still shows as opened.

No need to address that; it's an edge case that will likely be difficult to resolve and will have a high blast radius.

File: lua/nvim-tree/lib.lua (hunk 1 above)
Comment (Danie-1) [reply lua/nvim-tree/lib.lua:84]: Would a function to "apply function to all nodes in group" be more useful instead maybe?

**Comment by Danie-1** [comment]
I had a brief look at earning the bonus points but it looked like it would take me a little bit of time to figure what would be a suitable variable name to replace it (in some of the places). I can open a pull request for it at some point if noone else does it first.

**Comment by Danie-1** [comment]
Also I was expecting there might still be some edge cases this didn't handle but I was thinking maybe some of those edge cases might be caused by a similar issue to this one, where the state of each node in a group should be changed rather than a single node (but I'm not at all certain about that because I haven't found buggy edge cases or looked at the code for stuff).

File: lua/nvim-tree/lib.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree/lib.lua:84]: > Would a function to "apply function to all nodes in group" be more useful instead maybe?

See `NodeIterator`

You could use that here.

**Comment by alex-courtis** [comment]
> I had a brief look at earning the bonus points but it looked like it would take me a little bit of time to figure what would be a suitable variable name to replace it (in some of the places). I can open a pull request for it at some point if noone else does it first.

No worries, well out of scope.

**Comment by alex-courtis** [comment]
> Also I was expecting there might still be some edge cases this didn't handle but I was thinking maybe some of those edge cases might be caused by a similar issue to this one, where the state of each node in a group should be changed rather than a single node (but I'm not at all certain about that because I haven't found buggy edge cases or looked at the code for stuff).

That's fine. Folder/grouping is complex and we can only fix what we know about.

**Review by alex-courtis — APPROVED** [review]

**Pull request merged**